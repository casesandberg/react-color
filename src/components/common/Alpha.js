'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

import Checkboard from './Checkboard'

export class Alpha extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        alpha: {
          Absolute: '0px 0px 0px 0px',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0px 0px 0px 0px',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0px 0px 0px 0px',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0px 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0px 0px 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    }
  }

  componentWillUnmount() {
    this.unbindEventListeners()
  }

  handleChange = (e: any, skip: boolean) => {
    !skip && e.preventDefault()
    var container = this.refs.container
    var containerWidth = container.clientWidth
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset)

    var a
    if (left < 0) {
      a = 0
    } else if (left > containerWidth) {
      a = 1
    } else {
      a = Math.round(left * 100 / containerWidth) / 100
    }

    if (this.props.a !== a) {
      this.props.onChange({
        h: this.props.hsl.h,
        s: this.props.hsl.s,
        l: this.props.hsl.l,
        a: a,
        source: 'rgb',
      })
    }
  }

  handleMouseDown = (e: any) => {
    this.handleChange(e, true)
    window.addEventListener('mousemove', this.handleChange)
    window.addEventListener('mouseup', this.handleMouseUp)
  }

  handleMouseUp = () => {
    this.unbindEventListeners()
  }

  unbindEventListeners = () => {
    window.removeEventListener('mousemove', this.handleChange)
    window.removeEventListener('mouseup', this.handleMouseUp)
  }

  render(): any {
    var pointer = <div is="slider" />

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown }
            onTouchMove={ this.handleChange }
            onTouchStart={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    )
  }
}

export default Alpha
