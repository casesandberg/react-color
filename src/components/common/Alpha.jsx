'use strict';

var React = require('react');
var ReactCSS = require('reactcss');

var Checkboard = require('./Checkboard');

class Alpha extends ReactCSS.Component {

  constructor() {
    super();

    this.handleChange = this.handleChange.bind(this);
    this.handleMouseDown = this.handleMouseDown.bind(this);
    this.handleMouseUp = this.handleMouseUp.bind(this);
  }

  classes() {
    return {
      'default': {
        alpha: {
          Absolute: '0 0 0 0',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0 0 0 0',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0 0 0 0',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0 0 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    };
  }

  handleChange(e, skip) {
    !skip && e.preventDefault();
    var container = this.refs.container;
    var containerWidth = container.clientWidth;
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset);

    var a;
    if (left < 0) {
      a = 0;
    } else if (left > containerWidth) {
      a = 1;
    } else {
      a = Math.round(left * 100 / containerWidth) / 100;
    }

    if (this.props.a !== a) {
      this.props.onChange({ h: this.props.hsl.h, s: this.props.hsl.s, l: this.props.hsl.l, a: a });
    }
  }

  handleMouseDown(e) {
    this.handleChange(e, true);
    window.addEventListener('mousemove', this.handleChange);
    window.addEventListener('mouseup', this.handleMouseUp);
  }

  handleMouseUp() {
    window.removeEventListener('mousemove', this.handleChange);
    window.removeEventListener('mouseup', this.handleMouseUp);
  }

  render() {
    var pointer = <div is="slider" />;

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />;
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown } onTouchMove={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    );
  }
}

module.exports = Alpha;
