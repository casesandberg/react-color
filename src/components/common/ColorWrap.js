'use strict' /* @flow */

import React from 'react'
import merge from 'merge'
import isPlainObject from 'lodash/isplainobject'
import debounce from 'lodash/debounce'
import color from '../../helpers/color'
import shallowCompare from 'react-addons-shallow-compare'

export const ColorWrap = (Picker) => {
  class ColorPicker extends React.Component {
    shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1]);

    constructor(props: any) {
      super()

      this.state = merge(color.toState(props.color, 0), {
        visible: props.display,
      })

      this.debounce = debounce(function (fn: any, data: any) {
        fn(data)
      }, 100)
    }

    handleChange = (data: any) => {
      data = color.simpleCheckForValidColor(data)
      if (data) {
        var colors = color.toState(data, data.h || this.state.oldHue)
        this.setState(colors)
        this.props.onChangeComplete && this.debounce(this.props.onChangeComplete, colors)
        this.props.onChange && this.props.onChange(colors)
      }
    }

    componentWillReceiveProps(nextProps: any) {
      this.setState(merge(color.toState(nextProps.color, this.state.oldHue), {
        visible: nextProps.display,
      }))
    }

    render(): any {
      return <Picker {...this.props} {...this.state} onChange={ this.handleChange } />
    }
  }

  ColorPicker.defaultProps = {
    color: {
      h: 250,
      s: .50,
      l: .20,
      a: 1,
    },
  }

  return ColorPicker
}

export default ColorWrap
