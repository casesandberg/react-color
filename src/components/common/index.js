export { default as Alpha } from './Alpha'
export { default as Checkboard } from './Checkboard'
export { default as EditableInput } from './EditableInput'
export { default as Hue } from './Hue'
export { default as Raised } from './Raised'
export { default as Saturation } from './Saturation'
export { default as ColorWrap } from './ColorWrap'
export { default as Swatch } from './Swatch'
