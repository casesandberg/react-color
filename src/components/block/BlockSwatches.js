import React from 'react'
import reactCSS from 'reactcss'
import map from 'lodash/map'

import { Swatch } from '../common'

export const BlockSwatches = ({ colors, onClick, onSwatchHover }) => {
  const styles = reactCSS({
    'default': {
      swatches: {
        marginRight: '-10px',
      },
      swatch: {
        width: '22px',
        height: '22px',
        float: 'left',
        marginRight: '10px',
        marginBottom: '10px',
        borderRadius: '4px',
      },
      clear: {
        clear: 'both',
      },
    },
  })

  return (
    <div style={ styles.swatches }>
      { map(colors, c => (
        <Swatch
          key={ c }
          color={ c }
          style={ styles.swatch }
          onClick={ onClick }
          onHover={ onSwatchHover }
          focusStyle={{
            boxShadow: `0 0 4px ${ c }`,
          }}
        />
      )) }
      <div style={ styles.clear } />
    </div>
  )
}

export default BlockSwatches
