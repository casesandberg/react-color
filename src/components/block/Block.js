import React from 'react'
import PropTypes from 'prop-types'
import reactCSS from 'reactcss'
import merge from 'lodash/merge'
import debounce from 'lodash/debounce'
import color from '../../helpers/color'

import { ColorWrap, EditableInput, Checkboard } from '../common'
import BlockSwatches from './BlockSwatches'

export const Block = ({ onChange, onSwatchHover, hex, colors, width, triangle,
  inputDebounce, styles: passedStyles = {}, className = '' }) => {
  const transparent = hex === 'transparent'
  const handleChange = (hexCode, e) => {
    color.isValidHex(hexCode) && onChange({
      hex: hexCode,
      source: 'hex',
    }, e)
  }

  const styles = reactCSS(merge({
    'default': {
      card: {
        width,
        background: '#fff',
        boxShadow: '0 1px rgba(0,0,0,.1)',
        borderRadius: '6px',
        position: 'relative',
      },
      head: {
        height: '110px',
        background: hex,
        borderRadius: '6px 6px 0 0',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'relative',
      },
      body: {
        padding: '10px',
      },
      label: {
        fontSize: '18px',
        color: color.getContrastingColor(hex),
        position: 'relative',
      },
      triangle: {
        width: '0px',
        height: '0px',
        borderStyle: 'solid',
        borderWidth: '0 10px 10px 10px',
        borderColor: `transparent transparent ${ hex } transparent`,
        position: 'absolute',
        top: '-10px',
        left: '50%',
        marginLeft: '-10px',
      },
      input: {
        width: '100%',
        fontSize: '12px',
        color: '#666',
        border: '0px',
        outline: 'none',
        height: '22px',
        boxShadow: 'inset 0 0 0 1px #ddd',
        borderRadius: '4px',
        padding: '0 7px',
        boxSizing: 'border-box',
      },
    },
    'hide-triangle': {
      triangle: {
        display: 'none',
      },
    },
  }, passedStyles), { 'hide-triangle': triangle === 'hide' })

  return (
    <div style={ styles.card } className={ `block-picker ${ className }` }>
      <div style={ styles.triangle } />

      <div style={ styles.head }>
        { transparent && (
          <Checkboard borderRadius="6px 6px 0 0" />
        ) }
        <div style={ styles.label }>
          { hex }
        </div>
      </div>

      <div style={ styles.body }>
        <BlockSwatches colors={ colors } onClick={ handleChange } onSwatchHover={ onSwatchHover } />
        <EditableInput
          style={{ input: styles.input }}
          value={ hex }
          onChange={ debounce(handleChange, inputDebounce) }
        />
      </div>
    </div>
  )
}

Block.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  colors: PropTypes.arrayOf(PropTypes.string),
  triangle: PropTypes.oneOf(['top', 'hide']),
  styles: PropTypes.object,
  inputDebounce: PropTypes.number,
}

Block.defaultProps = {
  width: 170,
  colors: ['#D9E3F0', '#F47373', '#697689', '#37D67A', '#2CCCE4', '#555555',
    '#dce775', '#ff8a65', '#ba68c8'],
  triangle: 'top',
  styles: {},
  inputDebounce: 750,
}

export default ColorWrap(Block)
