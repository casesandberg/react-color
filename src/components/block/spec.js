/* global test, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme';

import Block from './Block'
import BlockSwatches from './BlockSwatches'
import { Swatch } from '../common'
import color from '../../helpers/color'

test('Block renders correctly', () => {
  const tree = renderer.create(
    <Block />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Block onChange events correctly', () => {
  const changeSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Block onChange={changeSpy} />
  )
  expect(changeSpy).toHaveBeenCalledTimes(0)
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate('click')

  expect(changeSpy).toHaveBeenCalled()
})

test('Block with onSwatchHover events correctly', () => {
  const hoverSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Block onSwatchHover={hoverSpy} />
  )
  expect(hoverSpy).toHaveBeenCalledTimes(0)
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate('mouseOver')

  expect(hoverSpy).toHaveBeenCalled()
})

test('Block `triangle="none"`', () => {
  const tree = renderer.create(
    <Block triangle="none" />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('BlockSwatches renders correctly', () => {
  const tree = renderer.create(
    <BlockSwatches colors={ ['#fff', '#999', '#000'] } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
