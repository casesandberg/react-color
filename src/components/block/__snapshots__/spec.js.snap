// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block \`triangle="hide"\` 1`] = `
<div
  className="block-picker "
  style={
    Object {
      "MozBorderRadius": "6px",
      "MozBoxShadow": "0 1px rgba(0,0,0,.1)",
      "OBorderRadius": "6px",
      "OBoxShadow": "0 1px rgba(0,0,0,.1)",
      "WebkitBorderRadius": "6px",
      "WebkitBoxShadow": "0 1px rgba(0,0,0,.1)",
      "background": "#fff",
      "borderRadius": "6px",
      "boxShadow": "0 1px rgba(0,0,0,.1)",
      "msBorderRadius": "6px",
      "msBoxShadow": "0 1px rgba(0,0,0,.1)",
      "position": "relative",
      "width": 170,
    }
  }
>
  <div
    style={
      Object {
        "borderColor": "transparent transparent #22194d transparent",
        "borderStyle": "solid",
        "borderWidth": "0 10px 10px 10px",
        "display": "none",
        "height": "0px",
        "left": "50%",
        "marginLeft": "-10px",
        "position": "absolute",
        "top": "-10px",
        "width": "0px",
      }
    }
  />
  <div
    style={
      Object {
        "MozBorderRadius": "6px 6px 0 0",
        "OBorderRadius": "6px 6px 0 0",
        "WebkitBorderRadius": "6px 6px 0 0",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "background": "#22194d",
        "borderRadius": "6px 6px 0 0",
        "display": "flex",
        "height": "110px",
        "justifyContent": "center",
        "msBorderRadius": "6px 6px 0 0",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "color": "#fff",
          "fontSize": "18px",
          "position": "relative",
        }
      }
    >
      #22194d
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "marginRight": "-10px",
        }
      }
    >
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#D9E3F0",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#D9E3F0"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#F47373",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#F47373"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#697689",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#697689"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#37D67A",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#37D67A"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#2CCCE4",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#2CCCE4"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#555555",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#555555"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#dce775",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#dce775"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#ff8a65",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#ff8a65"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#ba68c8",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#ba68c8"
        />
      </span>
      <div
        style={
          Object {
            "clear": "both",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <input
        id="rc-editable-input-4"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder={undefined}
        spellCheck="false"
        style={
          Object {
            "MozBorderRadius": "4px",
            "MozBoxShadow": "inset 0 0 0 1px #ddd",
            "OBorderRadius": "4px",
            "OBoxShadow": "inset 0 0 0 1px #ddd",
            "WebkitBorderRadius": "4px",
            "WebkitBoxShadow": "inset 0 0 0 1px #ddd",
            "border": "0px",
            "borderRadius": "4px",
            "boxShadow": "inset 0 0 0 1px #ddd",
            "boxSizing": "border-box",
            "color": "#666",
            "fontSize": "12px",
            "height": "22px",
            "msBorderRadius": "4px",
            "msBoxShadow": "inset 0 0 0 1px #ddd",
            "outline": "none",
            "padding": "0 7px",
            "width": "100%",
          }
        }
        value="#22194D"
      />
    </div>
  </div>
</div>
`;

exports[`Block renders correctly 1`] = `
<div
  className="block-picker "
  style={
    Object {
      "MozBorderRadius": "6px",
      "MozBoxShadow": "0 1px rgba(0,0,0,.1)",
      "OBorderRadius": "6px",
      "OBoxShadow": "0 1px rgba(0,0,0,.1)",
      "WebkitBorderRadius": "6px",
      "WebkitBoxShadow": "0 1px rgba(0,0,0,.1)",
      "background": "#fff",
      "borderRadius": "6px",
      "boxShadow": "0 1px rgba(0,0,0,.1)",
      "msBorderRadius": "6px",
      "msBoxShadow": "0 1px rgba(0,0,0,.1)",
      "position": "relative",
      "width": 170,
    }
  }
>
  <div
    style={
      Object {
        "borderColor": "transparent transparent #22194d transparent",
        "borderStyle": "solid",
        "borderWidth": "0 10px 10px 10px",
        "height": "0px",
        "left": "50%",
        "marginLeft": "-10px",
        "position": "absolute",
        "top": "-10px",
        "width": "0px",
      }
    }
  />
  <div
    style={
      Object {
        "MozBorderRadius": "6px 6px 0 0",
        "OBorderRadius": "6px 6px 0 0",
        "WebkitBorderRadius": "6px 6px 0 0",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "background": "#22194d",
        "borderRadius": "6px 6px 0 0",
        "display": "flex",
        "height": "110px",
        "justifyContent": "center",
        "msBorderRadius": "6px 6px 0 0",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "color": "#fff",
          "fontSize": "18px",
          "position": "relative",
        }
      }
    >
      #22194d
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "marginRight": "-10px",
        }
      }
    >
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#D9E3F0",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#D9E3F0"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#F47373",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#F47373"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#697689",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#697689"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#37D67A",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#37D67A"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#2CCCE4",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#2CCCE4"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#555555",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#555555"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#dce775",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#dce775"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#ff8a65",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#ff8a65"
        />
      </span>
      <span
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozBorderRadius": "4px",
              "OBorderRadius": "4px",
              "WebkitBorderRadius": "4px",
              "background": "#ba68c8",
              "borderRadius": "4px",
              "cursor": "pointer",
              "float": "left",
              "height": "22px",
              "marginBottom": "10px",
              "marginRight": "10px",
              "msBorderRadius": "4px",
              "outline": "none",
              "position": "relative",
              "width": "22px",
            }
          }
          tabIndex={0}
          title="#ba68c8"
        />
      </span>
      <div
        style={
          Object {
            "clear": "both",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <input
        id="rc-editable-input-1"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder={undefined}
        spellCheck="false"
        style={
          Object {
            "MozBorderRadius": "4px",
            "MozBoxShadow": "inset 0 0 0 1px #ddd",
            "OBorderRadius": "4px",
            "OBoxShadow": "inset 0 0 0 1px #ddd",
            "WebkitBorderRadius": "4px",
            "WebkitBoxShadow": "inset 0 0 0 1px #ddd",
            "border": "0px",
            "borderRadius": "4px",
            "boxShadow": "inset 0 0 0 1px #ddd",
            "boxSizing": "border-box",
            "color": "#666",
            "fontSize": "12px",
            "height": "22px",
            "msBorderRadius": "4px",
            "msBoxShadow": "inset 0 0 0 1px #ddd",
            "outline": "none",
            "padding": "0 7px",
            "width": "100%",
          }
        }
        value="#22194D"
      />
    </div>
  </div>
</div>
`;

exports[`BlockSwatches renders correctly 1`] = `
<div
  style={
    Object {
      "marginRight": "-10px",
    }
  }
>
  <span
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#fff",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "22px",
          "marginBottom": "10px",
          "marginRight": "10px",
          "msBorderRadius": "4px",
          "outline": "none",
          "position": "relative",
          "width": "22px",
        }
      }
      tabIndex={0}
      title="#fff"
    />
  </span>
  <span
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#999",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "22px",
          "marginBottom": "10px",
          "marginRight": "10px",
          "msBorderRadius": "4px",
          "outline": "none",
          "position": "relative",
          "width": "22px",
        }
      }
      tabIndex={0}
      title="#999"
    />
  </span>
  <span
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "MozBorderRadius": "4px",
          "OBorderRadius": "4px",
          "WebkitBorderRadius": "4px",
          "background": "#000",
          "borderRadius": "4px",
          "cursor": "pointer",
          "float": "left",
          "height": "22px",
          "marginBottom": "10px",
          "marginRight": "10px",
          "msBorderRadius": "4px",
          "outline": "none",
          "position": "relative",
          "width": "22px",
        }
      }
      tabIndex={0}
      title="#000"
    />
  </span>
  <div
    style={
      Object {
        "clear": "both",
      }
    }
  />
</div>
`;
